typescript/examples/langchain-privy-delegated-chatbot/chatbot.ts

```diff
@@ -0,0 +1,224 @@
+  erc20ActionProvider,
+  pythActionProvider,
+  cdpApiActionProvider,
+  PrivyEvmEmbeddedWalletProvider,
+  cdpWalletActionProvider,
+} from "../../agentkit/src";
+import { getLangChainTools } from "@coinbase/agentkit-langchain";
+import { HumanMessage } from "@langchain/core/messages";
+import { MemorySaver } from "@langchain/langgraph";
+import { createReactAgent } from "@langchain/langgraph/prebuilt";
+import { ChatOpenAI } from "@langchain/openai";
+import * as dotenv from "dotenv";
+import * as readline from "readline";
+import fs from "fs";
+
+dotenv.config();
+
+const WALLET_DATA_FILE = "wallet_data.txt";
+
+/**
+ * Validates that required environment variables are set
+ *
+ * @throws {Error} - If required environment variables are missing
+ * @returns {void}
+ */
+function validateEnvironment(): void {
+  const missingVars: string[] = [];
+
+  // Check required variables
+  const requiredVars = [
+    "OPENAI_API_KEY",
+    "PRIVY_APP_ID",
+    "PRIVY_APP_SECRET",
+    "PRIVY_WALLET_AUTHORIZATION_PRIVATE_KEY",
+    "PRIVY_DELEGATED_WALLET_ID", // Required for embedded wallet
+  ];
+
+  requiredVars.forEach(varName => {
+    if (!process.env[varName]) {
+      missingVars.push(varName);
+    }
+  });
+
+  // Exit if any required variables are missing
+  if (missingVars.length > 0) {
+    console.error("Error: Required environment variables are not set");
+    missingVars.forEach(varName => {
+      console.error(`${varName}=your_${varName.toLowerCase()}_here`);
+    });
+    process.exit(1);
+  }
+}
+
+// Add this right after imports and before any other code
+validateEnvironment();
+
+/**
+ * Initialize the agent with Privy Agentkit
+ *
+ * @returns Agent executor and config
+ */
+async function initializeAgent() {
+  try {
+    // Initialize LLM
+    const llm = new ChatOpenAI({
+      model: "gpt-4o-mini",
+    });
+
+    let walletProvider: PrivyEvmEmbeddedWalletProvider;
+
+    const networkId = process.env.NETWORK_ID || "base";
+
+    if (networkId?.includes("solana")) {
+      console.error("Embedded wallet for Solana not implemented in this example");
+      process.exit(1);
+    } else {
+      walletProvider = await PrivyWalletProvider.configureWithWallet({
+        appId: process.env.PRIVY_APP_ID!,
+        appSecret: process.env.PRIVY_APP_SECRET!,
+        authorizationPrivateKey: process.env.PRIVY_WALLET_AUTHORIZATION_PRIVATE_KEY,
+        walletId: process.env.PRIVY_DELEGATED_WALLET_ID,
+        networkId: "base",
+        walletType: "embedded",
+      });
+    }
+
+    // Initialize AgentKit
+    const agentkit = await AgentKit.from({
+      walletProvider,
+      actionProviders: [
+        wethActionProvider(),
+        pythActionProvider(),
+        walletActionProvider(),
+        erc20ActionProvider(),
+        cdpApiActionProvider({
+          apiKeyName: process.env.CDP_API_KEY_NAME as string,
+          apiKeyPrivateKey: process.env.CDP_API_KEY_PRIVATE_KEY as string,
+        }),
+        cdpWalletActionProvider({
```

Comment:
Please remove `cdpWalletActionProvider` from the list (but keep `cdpApiActionProvider`)